### ISSUE TITLE
Error during save if changes

### ISSUE THREAD

**Issue by Miger88** [PhilippC/keepass2android#1389]
Since a few days I get an error when I try to save any changes. 
System: Android 10 on a Samsung SM-G973F (S10)

Code: General Exception
Message: An error occurred sending the request.

Can you tell me why?

Same problem after reinstalling the android app, open the onedrive database and just try to synchronize without local changes

**Comment by PhilippC** [comment]
are you using the OneDrive app? You should better try Change database - Open file - OneDrive

**Comment by Miger88** [comment]
I have tried it with change database.I have removed the application from my phone with clearing data and cache before deinstallation. After reinstallation and choosing import the existing file from onedrive I got the same issue.--Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.Am 05.10.20, 10:52 schrieb PhilippC <notifications@github.com>:

   
  are you using the OneDrive app? You should better try Change database - Open file - OneDrive

**Comment by fritzmg** [comment]
Same problem here. I tried both with Open file - OneDrive and opening the `*.kdbx` file directly via the OneDrive App. Synchronising with OneDrive currently does not work at all, at least for me 🙁

**Comment by Miger88** [comment]
@fritzmg I am using KeePassDx now. Works also with OneDrive and other cloud systems.

**Comment by fritzmg** [comment]
Unfortunately KeePassDX (on Android) corrupted my KeePass file while saving to OneDrive, so I don't really want to use that software again 😁

**Comment by Miger88** [comment]
I have no problems with keepass DX since I am using it. I think the corruption is comming from onedrive. I had same issue one time. Thats why I manually make weekly backups to antoher cloud system. So I only can loose new/changed passwords from one week.

**Comment by GraphiteWingedBird** [comment]
I'm having the same/similar issue, but only after making a change on my phone and trying to synchronize back to OneDrive. If I open the database and then manually sync, there aren't issues, and AFAIK if I make a change to the database in OneDrive it will resync/reload to the phone fine. This issue is new for me and I haven't done much troubleshooting, but the only major change is I created some entries that store .JPG images. Doesn't seem like that would be an issue, but the timing has me suspicious. I'm thinking I'll try a few things this weekend to narrow down the issue since the error code isn't helpful

**Comment by fritzmg** [comment]
@Miger88 I did try again with KeePassDX, but the same issue occurred. When trying to save changes to a database stored in OneDrive, the database gets corrupted and cannot be opened anymore anywhere. KeePassDX has known issues with OneDrive and Dropbox anyway. Since KeePass2Android supported OneDrive natively via the API I expect less problems there - but alas for some reason it does not work.

**Comment by Miger88** [comment]
@GraphiteWingedBird 

A few days before the problems I have added some attachment files to my keepass db. Some PDF Files, some JPG's. 


I thik it's a problem only if attachments are used. Good idea. 

@fritzmg  Do you work with file attachments? Try it without attachments. May be its working then.

**Comment by fritzmg** [comment]
Yes, my keepass file has indeed multiple file attachments.

**Comment by GraphiteWingedBird** [comment]
My testing was hit or miss with attachments, seems to be working now after removing the entries with attachments from the database and then re-introducing them. My issue always stemmed from changing the database on Android (specifically new entries) and then being unable to Merge or Overwrite those when saving. However, it is working for the time being - so until I organically create a new entry on my phone, I'm not going to make changes. I would guess that it is has something to do with attachments, but I'm not sure what the trigger is beyond attachments.

Depending on your setup and how often you use attachments, you might be able to save those in a separate database specific to attachments - especially if the issue is only when adding/editing entries via Android. For me, the attachments are something I seldom need or look at, but have in case I need them. So if I have to separate them, I'll probably be fine creating a separate 'child database' that I can either open manually when needed or link via KeeAutoExec. If the issue is truly creating new entries in a database with attachments, I will very very rarely need the ability to do that on my phone, so that attachment child database can effectively just be an offline/local cached copy of the attachments I might need

**Comment by Miger88** [comment]
For me its not a problem anymore because I use KeepassDX now.

**Comment by PhilippC** [comment]
I expect the issues with attachment is that the file exceeds certain file limits. Please try with 1.09a (switch to https://play.google.com/apps/testing/keepass2android.keepass2android if you haven't received it from regular update channal yet) where this should be fixed.

**Comment by fritzmg** [comment]
> I expect the issues with attachment is that the file exceeds certain file limits. Please try with 1.09a (switch to https://play.google.com/apps/testing/keepass2android.keepass2android if you haven't received it from regular update channal yet) where this should be fixed.

@PhilippC yes, that seems to have fixed it for me! This is how I tested:

1. I switched to https://play.google.com/apps/testing/keepass2android.keepass2android
2. After waiting a while I reinstalled KeePass2Android so that Google Play installs `1.09a`.
3. I removed all previous database files.
4. I went to _Open file..._ » _OneDrive_ » _OneDrive (My Files)_, connected to my Microsoft account and chose my `kdbx` file.
5. I added a new test entry and saved the database.

Saving the database and synchronising back to OneDrive worked this time and I confirmed on my PC that the test entry was indeed synchronised to OneDrive 👍 

Oktoberfest Beer donation is on its way to you ;)

**Comment by GraphiteWingedBird** [comment]
Yep - I haven't donated since I first started using the app so I'll see if I can afford an Oktoberfest case of beer!

I'm not seeing the issue on 1.09a, and my hunch about the hit or miss issues is that if you don't open/view the files on your Android, or due to some other quirk, I think saving the database back to OneDrive didn't always cause the issue. My database itself was never large enough to cause the issue in https://github.com/PhilippC/keepass2android/issues/1036, but after viewing an attachment locally on the phone I'd have the issue merging back to OneDrive.

I definitely think I'd call this one closed, only improvement left would be some kind of error capture code specific to this issue considering the general exception wasn't super helpful although it did help me find this issue on GitHub

**Comment by WebworkrNet** [comment]
@PhilippC As I perceive the progress of the ticket, the problem revolves around attached files. If this is ultimately confirmed, I would ask that the title of the ticket be adjusted accordingly. A thread with meaningful, appropriate and specific title will facilitate later research and may prevent duplicate tickets. Thanks :-)

**Comment by Miger88** [comment]
> @PhilippC As I perceive the progress of the ticket, the problem revolves around attached files. If this is ultimately confirmed, I would ask that the title of the ticket be adjusted accordingly. A thread with meaningful, appropriate and specific title will facilitate later research and may prevent duplicate tickets. Thanks :-)

I have edited the title
